misc/scripts/library-coverage/compare-files-comment-pr.py

```diff
@@ -125,5 +193,49 @@ def compare_folders(folder1, folder2, output_file):
         out.write(return_md)
 
 
-comment_pr(sys.argv[1], sys.argv[2], sys.argv[3],
-           sys.argv[4], sys.argv[5], sys.argv[6])
+def get_previous_run_id(repo, run_id, pr_number):
+    """
+    Gets the previous run id for a given workflow run, considering that the previous workflow run needs to come from the same PR.
+    """
+
+    # Get branch and repo from run:
+    this_run = utils.subprocess_check_output(["gh", "api", "-X", "GET", "repos/" + repo + "/actions/runs/" + str(
+        run_id), "--jq", "{ head_branch: .head_branch, head_repository: .head_repository.full_name }"])
+
+    this_run = json.loads(this_run)
+    pr_branch = this_run["head_branch"]
+    pr_repo = this_run["head_repository"]
+
+    # Get all previous runs that match branch, repo and workflow name:
+    ids = utils.subprocess_check_output(["gh", "api", "-X", "GET", "repos/" + repo + "/actions/runs", "-f", "event=pull_request", "-f", "status=success", "-f", "name=\"" + artifacts_worflow_name + "\"", "--jq",
+                                        "[.workflow_runs.[] | select(.head_branch==\"" + pr_branch + "\" and .head_repository.full_name==\"" + pr_repo + "\") | { created_at: .created_at, run_id: .id}] | sort_by(.created_at) | reverse | [.[].run_id]"])
+
+    ids = json.loads(ids)
+    if ids[0] != int(run_id):
+        raise Exception("Expected to find " + str(run_id) +
+                        " in the list of matching runs.")
+
+    for previous_run_id in ids[1:]:
+        utils.subprocess_run(["gh", "run", "download", "--repo", repo,
```

Comment:
You may want to pull out a helper function for calling `gh run download`, since you do it a few times.